xxx-uv_link_t.md

```diff
@@ -0,0 +1,67 @@
+| Title  | Port core to uv_link_t      |
+|--------|-----------------------------|
+| Author | @indutny                    |
+| Status | DRAFT                       |
+| Date   | 2016-06-03                  |
+
+## 1. Proposal
+
+I propose to replace `StreamBase` with [`uv_link_t`][0], and port existing
+classes that inherit from `StreamBase`.
```

Comment:
We had a conflict in `StreamBase` that lead to the method `GetAsyncWrap()` and `GetObject()`. Will this play nicely with `AsyncWrap`, and allow the class to propagate like it does with `JSStream`?